### ISSUE TITLE
MP3 Gap

### ISSUE THREAD

**Issue by beamer159** [performous/performous#133]
A feature I would appreciate is an MP3 gap. Similar to how Video Gap starts the video past its beginning, some song files have non-music content that I would like to skip. It would be very useful to specify a track to start a certain amount into the song file.

**Comment by nieknooijens** [comment]
does ultrastar itself have this feature? 
if yes then we happily implement it.
if no, then it's better to not do it, because we want to keep compability with ultrastar itself.

**Comment by kiliankoe** [comment]
There's a `#GAP:15500` parameter, which sets an offset between the lyrics and the music. Is this what you're looking for?

**Comment by beamer159** [comment]
Changing the #GAP value will change the time when the lyrics start appearing on the screen. What I am looking for is a parameter to change how far into a music file Performous will start playing that file.

For example, let's say that I have a music file that has a minute of non-music content at the beginning of it. It would be convenient if there was a #SOUNDGAP parameter I could set, which would start a minute into the sound file, thereby skipping the non-music content.

It just seems odd that Performous files has a parameter to set when lyrics appear (#GAP) and a parameter to set when a video starts playing (#VIDEOGAP), but not a parameter to set when a sound file starts playing.

**Comment by nieknooijens** [comment]
can't you just press "enter" to skip instrumental stuff?
otherwise you could cut the mp3 to the desired length with audacity, it even lets you insert a cool fade-in effect to mask the cut.
http://sourceforge.net/projects/audacity/ (or apt-get install audacity)

**Comment by beamer159** [comment]
Yes, pressing Enter allows me to manually skip instrumentals to get to the lyrics. Ideally though, I would like to not have to do this. Therefore, I actually have been cutting my sound files to eliminate the content at the beginning of them. This seems like the best solution to my problem right now. Unfortunately, this is considerably more work than simply specifying a #SOUNDGAP value in the lyrics file.

**Comment by Baklap4** [comment]
The problem is not implementing it the problem is the sites who users use to get their songs from www.usdb.animux.de and ultrastar-es.org use a specific format for their text files. If we add the new parameter you personally have to fill in the mp3gap for all those files as no one gives them for you with a few songs it might not be a problem. But if you have a lot of songs like that it'll be a pain in the ass

**Comment by beamer159** [comment]
I thought many of the parameters are optional. For example, many of my text files do not have a #VIDEOGAP parameter because it is not needed. However, for the songs that are not synced with the video, I included #VIDEOGAP. It seems like a #MP3GAP could also be an optional parameter. Text files on www.usbd.animux.de would not be affected by the change since the parameter they are missing is optional. However, for those who want to skip the beginning of, say, Paparrazzi (https://www.youtube.com/watch?v=d2smz_1L2_0), they would be able to add the #MP3GAP parameter to the text file.

**Comment by Lord-Kamina** [comment]
I could swear Ultrastar has a #START parameter, or is it only for video?

On Thursday, May 7, 2015, Arjan Spieard <notifications@github.com> wrote:

> The problem is not implementing it the problem is the sites who users use
> to get their songs from www.usdb.animux.de and ultrastar-es.org use a
> specific format for their text files. If we add the new parameter you
> personally have to fill in the mp3gap for all those files as no one gives
> them for you with a few songs it might not be a problem. But if you have a
> lot of songs like that it'll be a pain in the ass
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/performous/performous/issues/133#issuecomment-99895924>
> .
>


-- 

-- 
  Gregorio Litenstein Goldzweig [image: glit_qr.png]  Interno de
Medicina  Facultad
de Medicina Clí­nica Alemana - Universidad del Desarrollo

   - Fono: +56 9 96343643
   - E-Mail: g.litenstein@gmail.com

**Comment by beamer159** [comment]
Good point. Looking around, the only documentation I can find on #START is here: http://www.ultrastarstuff.com/html/tutorialtxtfile.html. Unfortunately, the description is not helpful at all, so I do not know what it does, or if it is supported by Performous.

**Comment by Lord-Kamina** [comment]
I know for sure I've got some songs that use that tag. Can't test them now
since I'm at work but I could report later today.

On Thursday, May 7, 2015, beamer159 <notifications@github.com> wrote:

> Good point. Looking around, the only documentation I can find on #START is
> here: http://www.ultrastarstuff.com/html/tutorialtxtfile.html.
> Unfortunately, the description is not helpful at all, so I do not know what
> it does, or if it is supported by Performous.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/performous/performous/issues/133#issuecomment-99920782>
> .
>


-- 

-- 
  Gregorio Litenstein Goldzweig [image: glit_qr.png]  Interno de
Medicina  Facultad
de Medicina Clí­nica Alemana - Universidad del Desarrollo

   - Fono: +56 9 96343643
   - E-Mail: g.litenstein@gmail.com

**Comment by beamer159** [comment]
I added the #START parameter to a song file and played it in Performous: no change, the song played exactly the same. I then played the same song in Ultrastar Deluxe with the #START parameter: The song's sound file started playing 10 seconds in. This is the behavior I am looking for. That means that Ultrastar already has this feature, which means it should not be problematic to implement in Performous, no?

**Comment by basisbit** [comment]
in USDX, there is a #START and a #FINISH file parameter. they "cut" from where to where a song file is played. See https://github.com/UltraStar-Deluxe/USDX/blob/master/src/base/USong.pas#L1093 for supported header parameters.

**Comment by nieknooijens** [comment]
I've added it to the "future" milestone. Will look into it next week

**Comment by nieknooijens** [comment]
so basically:
#VIDEOGAP: shift video
#GAP: shift lyrics
#START:shift audio.

those names are confusing :confused:

**Comment by Baklap4** [comment]
>so basically:
#VIDEOGAP: shift video
#GAP: shift lyrics
#START:shift audio.

Correct, and yeah they are however that's how usdx uses/implemented them so if we want to stay compatiable then we should aswell (as optional though as just a few files have them)

**Comment by Baklap4** [comment]
TODO

* Add support for the `START` tag.
* Add support for the `END` tag

All other tags are found here: https://github.com/UltraStar-Deluxe/USDX/blob/master/src/base/UFiles.pas#L137-L175

**Comment by marwin89** [comment]
In official format every major game and tool developer agreed to use #START tag. https://usdx.eu/format.
I close this issue.

**Comment by Baklap4** [comment]
Might be good to keep this open or create a referenced issue from this so we do get to implement it :)

**Comment by marwin89** [comment]
> Might be good to keep this open or create a referenced issue from this so we do get to implement it :)

you're right, I didn't read closely "and thought you already implemented that. sorry 😅
